frame/dapps-staking/src/weights.rs

```diff
@@ -122,23 +139,30 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
 	// Storage: DappsStaking ContractEraStake (r:1 w:1)
 	// Storage: DappsStaking GeneralEraInfo (r:1 w:0)
 	fn claim_dapp() -> Weight {
-		(31_619_000 as Weight)
+		(43_000_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(5 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
 	// Storage: DappsStaking PalletDisabled (r:1 w:0)
 	// Storage: DappsStaking ForceEra (r:0 w:1)
 	fn force_new_era() -> Weight {
-		(2_815_000 as Weight)
+		(5_000_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
 	// Storage: DappsStaking PalletDisabled (r:1 w:1)
 	fn maintenance_mode() -> Weight {
-		(10_970_000 as Weight)
+		(18_000_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
+	// Storage: DappsStaking PalletDisabled (r:1 w:0)
+	// Storage: DappsStaking Ledger (r:1 w:1)
+	fn set_reward_destination() -> Weight {
+		(24_000_000 as Weight)
+			.saturating_add(T::DbWeight::get().reads(2 as Weight))
```

Comment:
Can this be optimized?
Do you need 2 reads or is 1 enough?

Reply:
not sure what you mean, 1 read is to check is pallet disabled, the other is to get ledger for the origin account.
Can I drop one of them?

Reply:
Oh my bad, it skipped my mind that it's a struct we need to modify - not just a simple entry.

Disregard the comment.